### DISCUSSION TITLE
How to represent a transfer switch for Utility and Generator, before more distribution panels.

### DISCUSSION THREAD

**Discussion by ArumBlack** [opendcim/openDCIM#1543] [Q&A] [answered]
How are we supposed to represent a transfer switch between utility/generator sources that then feeds more panels? I only see transfer switch capability at the CDU level, but I cant feed more power panels from a CDU? We have a 600KW generator to a transfer switch that then feeds a 1200A distibution panel that powers a pair of 100KVA UPS and some non-ups load panels. the UPS's also then feed some Panels. The ups is fine I just make it a panel and I can monitor it from a device template. I just need to know how to fit the Transfer switch into the power design and if it can be monitored some way that is a bonus but we have other systems for that....

Thankyou!

**Comment by alex001x** [comment]
a pdu is connected to a power panel but if you check the sts box you can indicate 2 power panels
STS mode = switch power panel
if you have several different parent sources type Electric supplier + generator + UPS inverter
consider as electrical panel
enter in opendcim generator and inverter as power panel

Alexandre.

**Reply by ArumBlack** [reply]
Yes, but in this case the transfer switch feeds more panels, and I cannot feed a panel from a CDU. I tried creating a device template, [Generic]ATS, with ATS checkecd, and assigning that template to the Meter device template on the panel, but it still only allows one source.
<img width="736" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/c5d8d7ef-4ff3-410c-889d-e718846f37bd">

<img width="331" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/3df10cf7-9a1a-48c7-9aaf-ad108c72f6b5">

The ATS in question are like this:

![PXL_20240327_144825406](https://github.com/opendcim/openDCIM/assets/120606523/834e2d2b-b4f2-4c9b-8893-e9ad996de332)

And Feed this panel:

![PXL_20240111_200956258](https://github.com/opendcim/openDCIM/assets/120606523/cd47b10c-7f8b-4d42-a126-cf1495e8855b)

**Reply by ArumBlack** [reply]
And the panel fed by the ATS in turn feeds more panels, and 2 100KVA UPS, which also feed a panel each. those UPS fed panels supply the individual circuits that go to the Racks and feed the CDU's that finnally supply the devices with power.

**Comment by alex001x** [comment]
Hi,
on the power panel side, the parent source is always unique, 
Opendcim does not yet allow you to add several electrical sources.
CDU and STS type devices are designed to power racks, their sources are always power panels
Sorry, in your example it is not possible to model it on opendcim

You can always create the first level power panels and connect them to a single parent for reports and electrical loads even if this is not the reality of your installation.

I don't think Scott and the team Opendcim are developing anything in this direction, 
nothing planned for version 24.01 on this subject

Alexandre.

**Comment by samilliken** [comment, marked as answer]
Correct, that is outside of the scope of what we focus on for openDCIM.

Scott

**Reply by ArumBlack** [reply]
Thank you very much. I can live with this, and thank you for the functionality you have provided!
